From b2c1cd58fda04daa92b6f69253fd2ff8bc7ce8f4 Mon Sep 17 00:00:00 2001
From: Mike Perry <mikeperry-git@torproject.org>
Date: Tue, 4 Dec 2012 16:29:24 -0800
Subject: [PATCH 07/27] Make Tor Browser exit when not launched from Vidalia

Turns out the Windows 7 UI encourages users to "dock" their Tor Browser app
for easy relaunch. If they manage to do this, we should fail closed rather
than opened. Hopefully they will get the hint and dock Vidalia instead.

This is an emergency fix for
https://trac.torproject.org/projects/tor/ticket/4192. We can do a better
localized fix w/ a translated alert menu later, if it seems like this might
actually be common.
---
 browser/base/content/browser.js |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
index 35664ec..bd2feed 100644
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -1093,6 +1093,20 @@ var gBrowserInit = {
     // setup simple gestures support
     gGestureSupport.init(true);
 
+    // If this is not a TBB profile, exit. 
+    // Solves https://trac.torproject.org/projects/tor/ticket/4192
+    var foundPref = false;
+    try {
+      var ignored = gPrefService.getCharPref("torbrowser.version");
+      foundPref = true;
+    } catch(e) {
+      //dump("No pref: "+e);
+    }
+    if(!foundPref) {
+      var appStartup = Components.classes["@mozilla.org/toolkit/app-startup;1"]
+                           .getService(Components.interfaces.nsIAppStartup);
+      appStartup.quit(3); // Force all windows to close, and then quit.
+    }
 
     if (uriToLoad && uriToLoad != "about:blank") {
       if (uriToLoad instanceof Ci.nsISupportsArray) {
-- 
1.7.5.4

